### PR TITLE
[CI] If the xcode path is not present, use the default.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

File: tools/devops/automation/templates/build/build.yml [hunk 1]
```diff
@@ -111,6 +111,18 @@ steps:
   timeoutInMinutes: 30
   enabled: true
 
+# was the current xcode deleted (has happened due to a bug in the clean step)? then set the default so 
+# that we can call make and generated the provisionator config files.
+- bash: |
+    XCODE_SELECT=$(xcode-select -p)
+    if [[ -d $XCODE_SELECT ]]; then
+      ehco "Using Xcode in path $XCODE_SELECT"
```
Comment (mandel-macaque) [inline tools/devops/automation/templates/build/build.yml:119]: Dyslexia:
```suggestion
      echo "Using Xcode in path $XCODE_SELECT"
```

**Review by dalexsoto — APPROVED** [review]

File: tools/devops/automation/templates/build/build.yml [hunk 2]
```diff
@@ -111,6 +111,18 @@ steps:
...
 
+# was the current xcode deleted (has happened due to a bug in the clean step)? then set the default so 
+# that we can call make and generated the provisionator config files.
+- bash: |
+    XCODE_SELECT=$(xcode-select -p)
+    if [[ -d $XCODE_SELECT ]]; then
+      echo "Using Xcode in path $XCODE_SELECT"
+    else
+      echo "Setting Xcode to point to the default location."
+      xcode-select -s "/Appliations/Xcode.app/Contents/Developer"
```
Comment (mandel-macaque) [inline tools/devops/automation/templates/build/build.yml:122]:
```suggestion
      xcode-select -s "/Applications/Xcode.app/Contents/Developer"
```

**Review by spouliot — APPROVED** [review]

**Comment by mandel-macaque** [comment]
Merging since we already saw it works and fixes issues for the broken bots.

**Pull request merged**